tests/Integration/inc/common/rocketGetPurgeUrls.php

```diff
@@ -0,0 +1,163 @@
+<?php
+namespace WP_Rocket\Tests\Integration\inc\common;
+
+use WP_Rocket\Tests\Integration\FilesystemTestCase;
+use Brain\Monkey\Functions;
+
+/**
+ * @covers ::rocket_clean_cache_theme_update
+ * @uses   ::rocket_clean_domain
+ *
+ * @group  Common
+ * @group  Purge
+ */
+class Test_RocketGetPurgeUrls extends FilesystemTestCase {
+	protected $path_to_test_data = '/inc/common/rocketGetPurgeUrls.php';
+
+	private $site_options = [
+		'stylesheet' => ''
+	];
+
+	private $page_for_posts;
+	private $posts_map   = [];
+	private $authors_map = [];
+	private $post_data   = [];
+
+	public function setUp() {
+		$this->set_permalink_structure( "/%postname%/" );
+		parent::setUp();
+
+		$this->create_posts( $this->config['urls']['posts'] );
+
+		$this->create_authors( $this->config['urls']['authors'] );
+
+		Functions\expect( 'get_rocket_option' )->withAnyArgs()->andReturnUsing( function( $option_name, $default=null ) {
```

Comment:
@engahmeds3ed why is this needed in here? Can you please explain me? Probably I am missing something

Reply:
mocking this function that is used here:

https://github.com/wp-media/wp-rocket/blob/e5d7a6446d518f950bcbd86aa92fe03ac9276d72/inc/common/purge.php#L103

Reply:
This is an integration test. We should try to bypass the mocking and set the system to run with our configuration.
For this one this should do the trick:
```
add_filter( 'pre_get_rocket_option_cache_purge_pages', return_our_value);

// RUN THE CODE	

remove_filter( 'pre_get_rocket_option_cache_purge_pages', return_our_value );
```

Reply:
Ok